x/lockup/keeper/store.go

```diff
@@ -184,7 +184,15 @@ func (k Keeper) GetAccountLockedLongerDuration(ctx sdk.Context, addr sdk.AccAddr
 	return combineLocks(notUnlockings, unlockings)
 }
 
-// GetAccountLockedLongerDurationNotUnlockingOnly Returns account locked with duration longer than specified.
+// GetAccountLockedDuration Returns account locked with a specific duration
+func (k Keeper) GetAccountLockedDuration(ctx sdk.Context, addr sdk.AccAddress, duration time.Duration) []types.PeriodLock {
+	// it does not matter started unlocking or not for duration query
+	unlockings := k.getLocksFromIterator(ctx, k.AccountLockIteratorDuration(ctx, true, addr, duration))
+	notUnlockings := k.getLocksFromIterator(ctx, k.AccountLockIteratorDuration(ctx, false, addr, duration))
```

Comment:
This comment doesn't need to be directly addressed in this PR: 

I'm wondering if there is a way to avoid these double iterations with false and true values? If only we could specify to the iterator to consider any values for the next X bytes in the prefix, that would prevent us from having to iterate multiple times

Reply:
Are they double iterating? It seems to me that they're not, since unlocking locks and locked locks don't have the same prefix and iterator only iterates over given prefix